### PR TITLE
chore: document internal vite options

### PR DESCRIPTION
**Pull request by AtofStryker**
- Closes N/A

### Additional details
Wanted to take the opportunity to document some of the internal vite options as well as their function for posterity. Hopefully the added docs in the readme reflect things appropriately 😅 .
### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by lmiller1990 — COMMENTED** [review]
Had no idea half of these existed - good to know, thanks for documenting.

I don't think these should be documented here, though. This is more for the `vite-dev-server` we publish on npm for CT users - these variables are exclusively internal for developing Cypress. I'm not sure where this should go, but likely not in `npm`.

File: npm/vite-dev-server/README.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ From there, we check the "framework" field to source or define any known vite tr
 
 We then merge the sourced config with the user's vite config, and layer on our own transforms, and provide this to a vite instance. The vite instance used to create a vite-dev-server, which is returned.
 
+
```
Comment (AtofStryker) [inline npm/vite-dev-server/README.md:50]:
```suggestion
```

**Review by lmiller1990 — APPROVED** [review]

**Review by mschile — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -409,6 +409,18 @@ Each package is responsible for building itself and testing itself and can do so
 | `test-integration` | Run all integration tests within the package; `exit 0` if N/A                                                                                            |
 | `test-watch`       | Run all unit tests in the package in watch mode                                                                                                          |
 
+#### Internal Vite Options
+When executing top or package level scripts, [Vite](https://vitejs.dev/) may be used to build/host parts of the application. This section is to serve as a general reference for these environment variables that may be leverage throughout the repository.
+###### `CYPRESS_INTERNAL_VITE_DEV`
+Set to `1` if wanting to leverage [vite's](https://vitejs.dev/guide/#command-line-interface) `vite dev` over `vite build` to avoid a full [production build](https://vitejs.dev/guide/build.html).
+###### `CYPRESS_INTERNAL_VITE_INSPECT`
```
[the remark this answers is not in the document]
Comment (AtofStryker) [reply CONTRIBUTING.md:416]: I think it just needs to be truthy, so I can probably add here that setting to `'1'` turns it on, similar to vite dev

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -409,6 +409,18 @@ Each package is responsible for building itself and testing itself and can do so
...
+###### `CYPRESS_INTERNAL_VITE_DEV`
+Set to `1` if wanting to leverage [vite's](https://vitejs.dev/guide/#command-line-interface) `vite dev` over `vite build` to avoid a full [production build](https://vitejs.dev/guide/build.html).
+###### `CYPRESS_INTERNAL_VITE_INSPECT` 
+Used internally to leverage [vite-plugin-inspect](https://github.com/antfu/vite-plugin-inspect) to view intermediary vite plugin state. The `CYPRESS_INTERNAL_VITE_DEV` is required for this to be applied correctly.
+###### `CYPRESS_INTERNAL_VITE_OPEN_MODE_TESTING` 
+Set to `true` when doing internal cy-in-cy type tests to access the Cypress instance from the parent frame. Please see the [E2E Open Mode Testing](./guides/e2e-open-testing.md) Guide.
+###### `CYPRESS_INTERNAL_VITE_APP_PORT` 
+Leveraged only when `CYPRESS_INTERNAL_VITE_DEV` is set to spawn the vite dev server for the app on the specified port. The default port is `3333`.
+###### `CYPRESS_INTERNAL_VITE_LAUNCHPAD_PORT` 
+Leveraged only when `CYPRESS_INTERNAL_VITE_DEV` is set to spawn the vite dev server for the launchpad on the specified port. The default port is `3001`.
```
[the remark this answers is not in the document]
Comment (AtofStryker) [reply CONTRIBUTING.md:423]: It should already be below it https://github.com/cypress-io/cypress/pull/21667/files/8c769a60ab75531ef7e87b37e7ccddea9a5c5f87#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R414

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (AtofStryker) [reply CONTRIBUTING.md:416]: added in 4767c6c7a8

File: CONTRIBUTING.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mschile) [reply CONTRIBUTING.md:423]: I was thinking more like a hierarchy but I'm fine with it this way too. I would suggest updating the `CYPRESS_INTERNAL_VITE_OPEN_MODE_TESTING` description to match the wording on these two though (i.e. "Leveraged only when `CYPRESS_INTERNAL_VITE_DEV is set..."

[the remark this answers is not in the document]
Comment (AtofStryker) [reply CONTRIBUTING.md:423]: @mschile how's something like https://github.com/cypress-io/cypress/pull/21667/commits/bd409fa3b6d443b848dd41284f370585d4dc1c5f?

**Pull request merged**